samples/f77/demo_ftnlib.cpp

```diff
@@ -81,38 +93,34 @@ extern "C" {
             string fin = string(file, lenfile);
             string fth = string(id, lenid);
             trmodel = string(transport, lentr);
-            delete _gas;
-            _gas = new IdealGasMix(fin, fth);
+            _sol = newSolution(fin, fth, trmodel);
+            _gas = std::dynamic_pointer_cast<IdealGasPhase>(_sol->thermo());
+            _kin = std::dynamic_pointer_cast<GasKinetics>(_sol->kinetics());
+            _trans = _sol->transport();
```

Comment:
The `dynamic_pointer_cast`s are unnecessary if you change the type of `_gas` to `shared_ptr<ThermoPhase>`, and similar with `_kin`.